### PR TITLE
added option to disable wordcount options

### PR DESCRIPTION
**Pull request by ghost**
`Site.Params.show-wordcount` (defaults to true for compatibility, ease of discovery)

I just don't really like the wordcount, and would rather keep my markup clean than hide it with css.

Should I edit the README as well?

### CODE REVIEW

File: layouts/partials/article-meta.html [hunk 1]
```diff
@@ -1,5 +1,7 @@
 			<p>
 				{{ if not (eq (.Date.Format "2006-01-02") "0001-01-01") }}<time>{{ .Date.Format "2006-01-02" }}</time>{{ end }}
+{{ if default true (index .Site.Params "show-wordcount") }}
```
Comment (simonmika) [inline layouts/partials/article-meta.html]: I think we need to be stringent with naming of params. It should probably be `ShowWordCount`.

**Review by simonmika — CHANGES_REQUESTED** [review]
Thanks, I have been using the CSS solution myself, so far. Feel free to add the necessary documentation to the README in this PR as well.

File: layouts/partials/article-meta.html [hunk 2]
```diff
@@ -1,5 +1,7 @@
 			<p>
 				{{ if not (eq (.Date.Format "2006-01-02") "0001-01-01") }}<time>{{ .Date.Format "2006-01-02" }}</time>{{ end }}
+{{ if default true (index .Site.Params "ShowWordCount") }}
 				<span class='article-length'>{{ .FuzzyWordCount }} words</span>
 				<span class='article-reading-time'>{{ .ReadingTime }} minutes</span>
-			</p>
\ No newline at end of file
+{{ end }}
+			</p>
```
Comment (ghost) [inline layouts/partials/article-meta.html:7]: My editor adds a newline to the end of the file. Is this a big deal for you?

**Comment by ghost** [comment]
My README changes introduce a merge conflict with #10, since they both append to the **Configuring** section. In my own fork I resolved the conflict by putting this one's additions second.

File: layouts/partials/article-meta.html [hunk 3]
```diff
@@ -1,6 +1,6 @@
 			<p>
 				{{ if not (eq (.Date.Format "2006-01-02") "0001-01-01") }}<time>{{ .Date.Format "2006-01-02" }}</time>{{ end }}
-{{ if default true (index .Site.Params "show-wordcount") }}
+{{ if default true (index .Site.Params "ShowWordCount") }}
```
Comment (simonmika) [inline layouts/partials/article-meta.html]: What about:
```
{{ if default true .Site.Params.showWordCount }}
```
I have been looking around and I have realized that currently all site parameters are all lower case: `.Site.Params.showwordcount`. I think I had some issues when using JSON but we should probably go for camel case (`showWordCount`) and not pascal case (`ShowWordCount`).

Comment (ghost) [reply layouts/partials/article-meta.html]: I did a test, I added `{{ jsonify Site.Params }}`. Turns out hugo lowercases all the Params keys before passing it to the theme. This wouldn't have come up for you as all of your keys are single words. So we need to use lowercase, and case in config files can be left up to the user. Also explains why it wasn't working (but I don't know how I missed that it wasn't working before making this PR!)

File: layouts/partials/article-meta.html (hunk 2 above)
Comment (simonmika) [reply layouts/partials/article-meta.html:7]: That is how it should be. My editor is still broken in this respect.

File: layouts/partials/article-meta.html [hunk 4]
```diff
@@ -1,5 +1,7 @@
 			<p>
 				{{ if not (eq (.Date.Format "2006-01-02") "0001-01-01") }}<time>{{ .Date.Format "2006-01-02" }}</time>{{ end }}
+{{ if default true (index .Site.Params "showwordcount") }}
```
Comment (simonmika) [inline layouts/partials/article-meta.html:3]: Does thins work with the documentation above?

Comment (simonmika) [inline layouts/partials/article-meta.html:3]: How about using `.Site.Params.showwordcount` instead of `index .Site.Params "showwordcount`?

File: README.md [hunk 5]
```diff
@@ -33,6 +33,13 @@ Add pages to the `main` menu.
  * Menu with infinitive levels
  * Uses HTML 5 structural elements
 
+## Configuring
+
+### Word Counts
+
+By default, every article is preceded by a word count and estimated reading time.
+To diable this globally, set `Params.showWordCount` to `false` in your site config.
```
Comment (simonmika) [inline README.md]: It spells: disable.

**Review by simonmika — CHANGES_REQUESTED** [review]
Almost there.

File: layouts/partials/article-meta.html (hunk 4 above)
Comment (ghost) [reply layouts/partials/article-meta.html:3]: Yes, because showWordCount is actually lowercased by hugo. Turns out it stopped working when I switched from hyphenated-words to camelCase because the keys are all lowercase

Comment (ghost) [reply layouts/partials/article-meta.html:3]: my understanding is that `index` returns null if the key is not set, whereas directly referencing it will raise an error. I can change this though if I'm wrong.

**Comment by ghost** [comment]
I'm losing track of the thread structure. Could you summarize outstanding issues as a response here?

**Comment by simonmika** [comment]
As we have sorted out the `index` issue we should be fine now.

**Review by simonmika — APPROVED** [review]
